src/main/java/hudson/remoting/Launcher.java

```diff
@@ -326,95 +407,38 @@ public void run() throws Exception {
         }
         workDirManager.setupLogging(internalDirPath, agentLog != null ? PathUtils.fileToPath(agentLog) : null);
 
-        if(auth!=null) {
-            final int idx = auth.indexOf(':');
-            if(idx<0)   throw new CmdLineException(null, "No ':' in the -auth option", null);
-            Authenticator.setDefault(new Authenticator() {
-                @Override public PasswordAuthentication getPasswordAuthentication() {
-                    return new PasswordAuthentication(auth.substring(0,idx), auth.substring(idx+1).toCharArray());
-                }
-            });
-        }
```

Comment:
The only use of `-auth` (and only in this entrypoint: changing the default authenticator. This was dead code, as the values of `-credentials` and/or `-proxyCredentials` are used to preemptively send an authentication header.